### PR TITLE
#635: List Command Subcommands as separate commands

### LINKED ISSUE #635: Command Subcommands should be listed as separate commands
The output of `COMMAND LIST` in the current implementation in DiceDB outputs `COMMAND`, but in Redis it outputs all the subcommand as 

```
COMMAND|COUNT
COMMAND|DOCS
COMMAND|GETKEYS
COMMAND|GETKEYSANDFLAGS
COMMAND|HELP
COMMAND|INFO
COMMAND|LIST
```

So, make the output for subcommand of `COMMAND` command same as Redis.

### PR DESCRIPTION
**Pull request by hgupta12**
Resolves #635 -  Command Subcommands should be listed as separate commands

## Description 
This PR adds the sub-commands of `COMMAND` to the list of commands displayed on `COMMAND LIST` command.

## Screenshots
image

### CODE REVIEW

File: internal/eval/commands.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ type KeySpecs struct {
 
 var (
 	DiceCmds = map[string]DiceCmdMeta{}
+	CommandSubCommands []string
```
Comment (KaviiSuri) [inline internal/eval/commands.go:21]: Can this be moved info the diceCmdMetadata struct for COMMAND?

Comment (hgupta12) [reply internal/eval/commands.go]: Yes, that sounds like a good idea. This  will allow us to easily add sub-commands for other commands too.

**Review by lucifercr07 — CHANGES_REQUESTED** [review]
@hgupta12 please add unit tests and integration tests.

**Comment by hgupta12** [comment]
> @hgupta12 please add unit tests and integration tests.

The output of `COMMAND LIST` is unordered and would keep changing as more commands are added. I am not sure how to add unit test for such a scenario. Can you please suggest how to go about it? Integration test already exists for the the command at integration_tests/commands/async/command_list_test.go.

**Review by JyotinderSingh — APPROVED** [review]
Thanks for these fixes @hgupta12! The changes look good.

**Pull request merged**